playwright/tests/fixtures/baseTest.ts

```diff
@@ -0,0 +1,77 @@
+import { test as base, _electron, Page, chromium, Browser, ElectronApplication } from '@playwright/test';
+import { downloadAndUnzipVSCode } from '@vscode/test-electron/out/download';
+export { expect } from '@playwright/test';
+import path from 'path';
+import { prebuiltWorkspacePath } from '../workspaces/workspace';
+import { FileExplorer } from '../page-objects/explorer-view';
+import { downloadVersion } from '../utils/download-version';
+
+export const runInBrowser = process.env.RUN_IN_BRWOSER ? true : false;
+
+export const test = base.extend<{ workspace: string; page: Page }>({
+  workspace: prebuiltWorkspacePath,
+  page: async ({ workspace }, take) => {
+    const { page, browser, electronApp } = await appWithPage(workspace);
+    await initialize(page);
+    await take(page);
+    if (browser) {
+      await browser.close();
+      return;
+    }
+    if (electronApp) {
+      if (test.info().status === 'failed') {
+        const tracePath = test.info().outputPath('trace.zip');
+        const screenshotPath = test.info().outputPath('screenshot.png');
+        await page.context().tracing.stop({ path: tracePath });
+        await page.screenshot({ path: screenshotPath });
+        test.info().attachments.push({ name: 'trace', path: tracePath, contentType: 'application/zip' });
+        test.info().attachments.push({ name: 'screenshot', path: tracePath, contentType: 'image/png' });
+      }
+      await electronApp.close();
+    }
+  }
+});
+
+const appWithPage = async (workspace: string): Promise<{ page: Page; browser?: Browser; electronApp?: ElectronApplication }> => {
+  if (runInBrowser) {
```

Comment:
a little bit stragne to check this here return browser and the electronApp here as optional and then check it again in the test. why not check it in the test and then do also the intitalize and take for each, maybe even directly in the browser and electron functions.